thunder/distributed/transforms/fsdp_v2.py

```diff
@@ -57,12 +65,166 @@ def __call__(self, bsym: BoundSymbol) -> VISIT_TYPE:
         return VISIT_TYPE.REPLACE
 
 
-@dataclass(frozen=True)
-class FSDPTraceTransform(Transform):
+# When the user calls fsdp(jitted_module), this function does the following
+# - It transforms the ThunderModule jitted_module, materializing and sharding the parameters as `overrides`
+#   in the ThunderModule.
+# - While doing that, it leaves the original user module alone.
+# - It then registers an early transform (callback that runs before prologue is executed) that transforms the
+#   prologue and compute trace.
+#
+# Note that for doing so, there are a few constraints / caveats:
+# - We do not have prologues/compute traces when we transform the module.
+# - We need to record the info from the module transformations because a later transform might modify the module further.
+
+
+class FSDPTransform(Transform):
     sharded_params: dict[str, Any]
     process_group: ProcessGroup
     shared_params_name: dict[str, str]
 
+    def __init__(
+        self,
+        device: torch.device | None = None,
+        broadcast_from: int | None = None,
+        sharding_strategy: FSDPType = FSDPType.ZERO2,
+        bucketing_strategy: FSDPBucketingStrategy = FSDPBucketingStrategy.NONE,
+    ):
+        self.device = device
+        self.broadcast_from = broadcast_from
+        self.sharding_strategy = sharding_strategy
+        self.bucketing_strategy = bucketing_strategy
```

Comment:
nit-picking: I might prefer explicit init of these three here, and check the size / type of them in their callsite.
```suggestion
        self.bucketing_strategy = bucketing_strategy
        self.sharded_params: dict[str, Any] = {}
        self.process_group: ProcessGroup | None = None
        self.shared_params_name: dict[str, str] = {}
```